go/conversation/view_definition.py

```diff
@@ -574,11 +577,32 @@ def post(self, request, conversation):
 
     @check_action_is_enabled
     def perform_action(self, request, conversation, action_data):
-        self.action.delegate_perform_action(action_data)
-        messages.info(request, 'Action successful: %s!' % (
-            self.action.action_display_name,))
+        """Sends the action to the correct function depending on whether the
+        action should be scheduled or not."""
+        if action_data.get('scheduled_datetime') is None:
+            self.action.perform_action(action_data)
+            messages.info(request, 'Action successful: %s!' % (
+                self.action.action_display_name,))
+        else:
+            self.schedule_action(action_data, conversation)
+            messages.info(request, 'Action successfully scheduled: %s!' % (
+                self.action.action_display_name,))
         return self._action_done(request, conversation)
 
+    def schedule_action(self, action_data, conversation):
+        scheduled_for = action_data.pop('scheduled_datetime')
+        action_data.pop('display')
```

Comment:
Ah, it's for displaying the details of the action to the user during action confirmation. That means that ignoring it here is the correct thing to do.